src/main/java/com/javadeobfuscator/deobfuscator/transformers/normalizer/DuplicateRenamer.java

```diff
@@ -0,0 +1,458 @@
+/*
+ * Copyright 2016 Sam Sun <me@samczsun.com>
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.javadeobfuscator.deobfuscator.transformers.normalizer;
+
+import java.util.*;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.atomic.AtomicInteger;
+
+import org.objectweb.asm.Type;
+import org.objectweb.asm.tree.ClassNode;
+import org.objectweb.asm.tree.FieldNode;
+import org.objectweb.asm.tree.MethodNode;
+
+import com.javadeobfuscator.deobfuscator.config.TransformerConfig;
+import com.javadeobfuscator.deobfuscator.executor.Context;
+import com.javadeobfuscator.deobfuscator.executor.defined.types.JavaClass;
+import com.javadeobfuscator.deobfuscator.executor.defined.types.JavaMethod;
+import com.javadeobfuscator.deobfuscator.executor.providers.DelegatingProvider;
+import com.javadeobfuscator.deobfuscator.utils.ClassTree;
+
+@TransformerConfig.ConfigOptions(configClass = DuplicateRenamer.Config.class)
+public class DuplicateRenamer extends AbstractNormalizer<DuplicateRenamer.Config>
```

Comment:
Why is this needed? Won't the regular Class/Field/Method normalizers achieve the same result of distinguishing members?

Reply:
This is an alternative (for me at least) to preserve the original class/method/field names as much as possible while still making it possible to extract the files.
With the introduction of the config, allowing certain files to be excluded may solve this.
For the unconditional switch, its only used to fix ZKM's string encryption at <clinit>, as it sometimes can't be decompiled without that. A suggestion is to include the unconditionalswitch as part of ZKM string decryptor.